File: Buoi2/Python/Bubble.py
# Hàm Bubble Sort
def bubbleSort(arr):
    n = len(arr)
    for i in range(n - 1):
        swapped = False
        for j in range(n - 1 - i): # Điều chỉnh vòng lặp để tránh so sánh phần tử đã được sắp xếp
            if arr[j] > arr[j + 1]: # Hoán đổi
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                swapped = True
        # Nếu không có hoán đổi nào, mảng đã được sắp xếp
        if not swapped:
            break

arr = [5, 2, 9, 1, 5]
bubbleSort(arr)
print("Ket qua sau Bubble Sort:", arr)

File: Buoi2/Julia/Bubble.jl
# Hàm Bubble Sort
function bubbleSort(arr)
    n = length(arr)
    for i in 1:n-1
        swapped = false
        for j in 1:n-1-i  # Điều chỉnh vòng lặp để tránh so sánh phần tử đã được sắp xếp
            if arr[j] > arr[j + 1]  # Hoán đổi
                temp = arr[j]
                arr[j] = arr[j + 1]
                arr[j + 1] = temp
                swapped = true
            end
        end
        # Nếu không có hoán đổi nào, mảng đã được sắp xếp
        if !swapped
            break
        end
    end
end

arr = [5, 2, 9, 1, 5]
bubbleSort(arr)
println("Ket qua sau Bubble Sort:", arr)
